tst/EnergyPlus/unit/VariableSpeedCoils.unit.cc

```diff
@@ -3019,57 +3025,82 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_ContFanCycCoil_Test)
-        state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).AirMassFlowRate;
+    vsCoolingCoil.AirMassFlowRate = vsCoolingCoil.MSRatedAirMassFlowRate(1) * PartLoadFrac;
+    state->dataVariableSpeedCoils->LoadSideMassFlowRate = vsCoolingCoil.AirMassFlowRate;
+    state->dataLoopNodes->Node(vsCoolingCoil.AirInletNodeNum).MassFlowRate = vsCoolingCoil.AirMassFlowRate;
     // run the coil
     VariableSpeedCoils::CalcVarSpeedCoilCooling(
         *state, DXCoilNum, fanOp, SensLoad, LatentLoad, compressorOp, PartLoadFrac, OnOffAirFlowRatio, SpeedRatio, SpeedCal);
     // check coil air outlet conditions
-    EXPECT_NEAR(state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).OutletAirDBTemp, 5.23333, 0.00001);
-    EXPECT_NEAR(state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).OutletAirHumRat, 0.00810, 0.00001);
-    EXPECT_NEAR(state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).OutletAirEnthalpy, 25597.58, 0.01);
+    EXPECT_NEAR(vsCoolingCoil.OutletAirDBTemp, 5.23333, 0.00001);
+    EXPECT_NEAR(vsCoolingCoil.OutletAirHumRat, 0.00810, 0.00001);
+    EXPECT_NEAR(vsCoolingCoil.OutletAirEnthalpy, 25597.58, 0.01);
+    EXPECT_NEAR(vsCoolingCoil.Power, 785.588, 0.001);
+    EXPECT_NEAR(vsCoolingCoil.QSource, 3915.355, 0.001);
+    EXPECT_NEAR(vsCoolingCoil.QLoadTotal, 3182.143, 0.001);
+    ;
+    // test 3: dx cooling coil unavailable, compressor is On and PLR > 0
+    // run init the coil to reset coil air inlet and outlet conditions
+    VariableSpeedCoils::InitVarSpeedCoil(*state, DXCoilNum, SensLoad, LatentLoad, fanOp, OnOffAirFlowRatio, SpeedRatio, SpeedCal);
+    compressorOp = HVAC::CompressorOp::On;
+    PartLoadFrac = 0.1;
+    // reset the cooling coil availability schedule to AlwaysOff
+    vsCoolingCoil.availSched = Sched::GetScheduleAlwaysOff(*state);
+    // set coil inlet condition
+    vsCoolingCoil.InletAirDBTemp = 24.0;
+    vsCoolingCoil.InletAirHumRat = 0.009;
+    vsCoolingCoil.InletAirEnthalpy = Psychrometrics::PsyHFnTdbW(24.0, 0.009);
+    // set coil inlet air flow rate to speed 1 times PLR
+    vsCoolingCoil.AirMassFlowRate = vsCoolingCoil.MSRatedAirMassFlowRate(1) * PartLoadFrac;
+    state->dataVariableSpeedCoils->LoadSideMassFlowRate = vsCoolingCoil.AirMassFlowRate;
+    state->dataLoopNodes->Node(vsCoolingCoil.AirInletNodeNum).MassFlowRate = vsCoolingCoil.AirMassFlowRate;
+    // run the coil
+    VariableSpeedCoils::CalcVarSpeedCoilCooling(
+        *state, DXCoilNum, fanOp, SensLoad, LatentLoad, compressorOp, PartLoadFrac, OnOffAirFlowRatio, SpeedRatio, SpeedCal);
+    // check coil air outlet conditions to the initial air condition, which is all zero
+    EXPECT_EQ(vsCoolingCoil.OutletAirDBTemp, 0.0);
+    EXPECT_EQ(vsCoolingCoil.OutletAirHumRat, 0.0);
+    EXPECT_EQ(vsCoolingCoil.OutletAirEnthalpy, 0.0);
+    EXPECT_EQ(vsCoolingCoil.Power, 0.0);
+    EXPECT_EQ(vsCoolingCoil.QSource, 0.0);
+    EXPECT_EQ(vsCoolingCoil.QLoadTotal, 0.0);
 }
```

Comment:
Modified an existing unit test to test report variables when the VS DX cooling coil is off by changing the availability schedule to off. The DX coil air outlet conditions were reset to the default value of `0.0`, and the performance output variables were reset to `0.0`.